Makefile

```diff
@@ -1,125 +1,133 @@
 
-build_deps:
+.PHONY: build_deps
+build_deps:  ## build container images of dependnecies (FMS, ESMF, SerialBox)
 	docker build -f $(DOCKERFILE) -t $(MPI_IMAGE) $(BUILD_ARGS) --target fv3gfs-mpi .
 	docker build -f $(DOCKERFILE) -t $(FMS_IMAGE) $(BUILD_ARGS) --target fv3gfs-fms .
 	docker build -f $(DOCKERFILE) -t $(ESMF_IMAGE) $(BUILD_ARGS) --target fv3gfs-esmf .
 	docker build -f $(DOCKERFILE) -t $(SERIALBOX_IMAGE) $(BUILD_ARGS) --target fv3gfs-environment-serialbox .
 
-push_deps:
+.PHONY: push_deps
+push_deps:  ## push container images of dependencies to GCP
 	docker push $(MPI_IMAGE)
 	docker push $(FMS_IMAGE)
 	docker push $(ESMF_IMAGE)
 	docker push $(SERIALBOX_IMAGE)
 
-pull_deps:
+.PHONY: pull_deps
+pull_deps:  ## pull container images of dependnecies from GCP (for faster builds)
 	docker pull $(MPI_IMAGE)
 	docker pull $(FMS_IMAGE)
 	docker pull $(ESMF_IMAGE)
 	docker pull $(SERIALBOX_IMAGE)
 
-build_debug: build_environment
-	COMPILED_TAG_NAME=debug COMPILE_OPTION="REPRO=\\\nDEBUG=Y" $(MAKE) build
-
-build_coverage: build_environment
-	COMPILED_TAG_NAME=gcov COMPILED_IMAGE=$(GCR_URL)/$(COMPILE_TARGET):gcov \
-	COMPILE_OPTION="OPENMP=\\\nREPRO=\\\nDEBUG=Y\\\nGCOV=Y" $(MAKE) build
+.PHONY: enter
+enter:  ## run and enter production container for development
+	docker run --rm \
+	       	-v $(shell pwd)/FV3:/FV3 $(OTHER_MOUNTS) \
+		-w /FV3 -it $(COMPILED_IMAGE) bash
 
-enter:
-	docker run --rm $(MOUNTS) -w /FV3 -it $(COMPILED_IMAGE) bash
+.PHONY: enter_serialize
+enter_serialize:  ## run and enter serialization container for development
+	docker run --rm \
+		-v $(shell pwd)/FV3:/FV3/original $(OTHER_MOUNTS) \
+		-w /FV3 -it $(SERIALIZE_IMAGE) bash
 
-enter_serialize:
-	MOUNTS="$(MOUNTS_SERIALIZE)" COMPILED_IMAGE="$(SERIALIZE_IMAGE)" $(MAKE) enter
-
-compile_dev: build_compiled
-	docker run --rm $(MOUNTS) -w /FV3 -it $(COMPILED_IMAGE) bash -c "make libs && make fv3.exe"
-
-test:
+.PHONY: test
+test:  ## run tests (set COMPILED_TAG_NAME to override default)
 	pytest tests/pytest --capture=no --verbose --refdir $(shell pwd)/tests/pytest/reference/circleci --image_version $(COMPILED_TAG_NAME)
 
-update_circleci_reference: test
+.PHONY: update_circleci_reference
+update_test_reference: test  ## update md5 checksums for regression tests
 	cd tests/pytest && bash set_reference.sh $(COMPILED_TAG_NAME)-serialize $(shell pwd)/reference/circleci
 
-# 32bit options don't currently build, fix these when issue #4 is fixed.
+.PHONY: clean
+clean:  ## cleanup source tree and test output
+	(cd FV3 && make clean)
+	$(RM) -f inputdata
+	$(RM) -rf tests/pytest/output/*
+
+# TODO 32bit options don't currently build, fix these when issue #4 is fixed.
 #test_32bit:
 #	COMPILED_TAG_NAME=32bit $(MAKE) test
 #
 #build_32bit: build_environment
 #	COMPILED_TAG_NAME=32bit COMPILE_OPTION=32BIT=Y $(MAKE) build
-
-dev_serialize: # TODO: use run_docker -- string form of command is not translating correctly
```

Comment:
Is it OK to delete this make target?

Reply:
My understanding is that it has not been updated to the new structure of how serialization is done (e.g. /FV3/original). @rheacangeo ?

Reply:
Yeah this is a few generations out of date... I forgot about it as the serialization 'framework' has been stable for a long time so I haven't needed to iterate on the statements much when I make changes. This could potentially start being an issue when we revisit these, and we might want to add it back, but it's still assuming the serialized files go to /Serialize.  Now it might make sense to add something like this to the Makefile in test/serialized_test_data_generation, but that is out of the scope of this PR.